integration/spark/src/test/scala/org/apache/spark/carbondata/query/TestFilterReordering.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.carbondata.query
+
+import java.util
+
+import org.apache.spark.sql.{CarbonEnv, CarbonThreadUtil}
+import org.apache.spark.sql.test.util.QueryTest
+import org.scalatest.BeforeAndAfterAll
+
+import org.apache.carbondata.core.constants.CarbonCommonConstants
+import org.apache.carbondata.core.scan.expression.{ColumnExpression, Expression, LiteralExpression}
+import org.apache.carbondata.core.scan.expression.conditional.EqualToExpression
+import org.apache.carbondata.core.scan.expression.logical.{AndExpression, OrExpression}
+import org.apache.carbondata.core.scan.expression.optimize.ExpressionOptimizer
+
+class TestFilterReordering extends QueryTest with BeforeAndAfterAll {
+
+  override protected def beforeAll(): Unit = {
+    sql("drop table if exists filter_reorder")
+    sql("create table filter_reorder(one string, two string, three string, four int, " +
+      "five int) stored as carbondata")
+  }
+
+  test("Test filter reorder with various conditions") {
+    checkOptimizer("(four = 11 and two = 11) or (one = 11)",
```

Comment:
why pass the filter expression as string instead of Expression object?? No need for translation.

Refer: https://github.com/apache/carbondata/pull/3902/files#diff-ddfc2bc65b7f1055d4ae72fdfcdad5418e44373a6391fc348568b9a8bee506f6